### PR TITLE
fix: correctly handle Substrait windows with rows bounds (and validate executability of test plans)

### PR DESCRIPTION
**Pull request by Blizzara**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Consuming Substrait Windows with ROWS bounds other than current or unbounded would fail, not at planning time but only at execution time, with `Error: Internal("Rows should be Uint")`.

This wasn't being caught by roundtrip tests for two reasons: 1) no test specified bounds, and 2) we didn't actually execute any of the plans after creating them. This can be seen in the three commits of this PR: the first commit adding a test case by itself passes all checks, the second makes it fail, and the third fixes it again.

## What changes are included in this PR?

Fix Substrait consumer to produce uint64 bounds instead of int64. 
Make Substrait roundtrip tests actually execute the post-roundtrip plan to confirm it's valid. 
Fix existing test cases where that wasn't working (notably move interval tests into producer.rs since arrow_cast doesn't support casting interval types).

## Are these changes tested?

Yes, through new and improved existing unit tests.

### CODE REVIEW

File: datafusion/substrait/tests/testdata/data.csv [hunk 1]
```diff
@@ -1,3 +1,3 @@
 a,b,c,d,e,f
-1,2.0,2020-01-01,false,4294967296,'a'
-3,4.5,2020-01-01,true,2147483648,'b'
\ No newline at end of file
+1,2.0,2020-01-01,false,4294967295,'a'
```
Comment (alamb) [inline datafusion/substrait/tests/testdata/data.csv:2]: Why was this changed?

File: datafusion/substrait/tests/cases/roundtrip_logical_plan.rs [hunk 2]
```diff
@@ -523,24 +523,6 @@ async fn roundtrip_arithmetic_ops() -> Result<()> {
     Ok(())
 }
 
-#[tokio::test]
-async fn roundtrip_interval_literal() -> Result<()> {
-    roundtrip(
-        "SELECT g from data where g = arrow_cast(INTERVAL '1 YEAR', 'Interval(YearMonth)')",
```
Comment (alamb) [inline datafusion/substrait/tests/cases/roundtrip_logical_plan.rs:529]: Are you referring to the fact that this would actually error at runtime:
```sql
> select arrow_cast(INTERVAL '1 YEAR', 'Interval(YearMonth)');
This feature is not implemented: Unsupported CAST from Interval(MonthDayNano) to Interval(YearMonth)
```

File: datafusion/substrait/tests/testdata/data.csv (hunk 1 above)
Comment (Blizzara) [reply datafusion/substrait/tests/testdata/data.csv:2]: The old value was 1 larger than what fits into uint64, so executing against this file failed.

I didn’t see any reason why the over-large value would be useful, so this brings it down into the range that actually fits into uint64

File: datafusion/substrait/tests/cases/roundtrip_logical_plan.rs (hunk 2 above)
Comment (Blizzara) [reply datafusion/substrait/tests/cases/roundtrip_logical_plan.rs:529]: Yup

File: datafusion/substrait/tests/testdata/data.csv (hunk 1 above)
Comment (alamb) [reply datafusion/substrait/tests/testdata/data.csv:2]: I think you mean `Int32` 🤔 

```sql
> select 4294967295::int;
Arrow error: Cast error: Can't cast value 4294967295 to type Int32
> select 4294967295::bigint;
+-------------------+
| Int64(4294967295) |
+-------------------+
| 4294967295        |
+-------------------+
1 row(s) fetched.
Elapsed 0.002 seconds.
```

**Review by alamb — APPROVED** [review]
Thanks @Blizzara

File: datafusion/substrait/tests/testdata/data.csv (hunk 1 above)
Comment (Blizzara) [reply datafusion/substrait/tests/testdata/data.csv:2]: Ah yes, I misremembered. It’s int32 as specified here https://github.com/apache/datafusion/blob/2aa57386c12e57d2fe865360f3eead358cc237f7/datafusion/substrait/tests/cases/roundtrip_logical_plan.rs#L1125. Maybe the original number at some point had some meaning to test the int32/int64 difference, but I don’t see what purpose it’s serve in the substrait roundtrip tests 😅

**Comment by alamb** [comment]
Thanks again!
